thewalrus/decompositions.py

```diff
@@ -202,6 +202,17 @@ def takagi(A, svd_order=True):
         vals, U = takagi(Amr, svd_order=svd_order)
         return vals, U * np.exp(1j * phi / 2)
 
+    # If the matrix is diagonal, Takagi decomposition is easy
+    if np.allclose(A, np.diag(np.diag(A)), rtol=1e-16):
+        d = np.diag(A)
+        U = np.diag(np.exp(1j * 0.5 * np.angle(d)))
+        l = np.abs(d)
+        l = np.sort(l)
+        U = U[np.argsort(l)]
+        if svd_order:
+            return l[::-1], U[:, ::-1]
+        return l, U
```

Comment:
you can do the same thing on lines 218-220 as well if it helps, but I think this should suffice.
```suggestion
        return (l[::-1], U[:, ::-1]) if svd_order else (l, U)
```